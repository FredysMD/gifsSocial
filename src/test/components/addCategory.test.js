import React from 'react';
import { shallow } from 'enzyme';

import  AddCategory  from '../../components/AddCategory';


describe('Test to AddCategory component', () => {
    
    //const setCategories = () => {};
    const setCategories = jest.fn();
    
    let wrapper = shallow( <AddCategory setCategories = { setCategories } /> ); // for have the snippets

    beforeEach( () => {

        jest.clearAllMocks();

        wrapper = shallow( <AddCategory setCategories = { setCategories } /> );

    });
    
    test('should have addCategory component', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('should change field input ', () => {
        
        const input = wrapper.find('input');
        
        const value = 'Messi';

        input.simulate('change', { target: { value } });

        const p = wrapper.find('p').text().trim();

        expect( p ).toBe( value );
        
    });
    
    test('should not post the info in form', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} }); // function empty; equal to () => {}

        expect( setCategories ).not.toHaveBeenCalled();

    });
    
    test('should call setCategory and clean InputValue tag', () => {
        
        const value = 'Mary';
        wrapper.find('input').simulate('change', { target : { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        const inputV = wrapper.find('input');

        expect( setCategories ).toHaveBeenCalled();
        expect( inputV.prop('value') ).toBe('');
        
    });
        
})
