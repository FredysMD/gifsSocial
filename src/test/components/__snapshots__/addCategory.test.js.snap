// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to AddCategory component should have addCategory component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
